src/main/kotlin/uk/gov/dluhc/printapi/service/CertificateRemovalDateResolver.kt

```diff
@@ -0,0 +1,58 @@
+package uk.gov.dluhc.printapi.service
+
+import org.springframework.stereotype.Component
+import uk.gov.dluhc.printapi.client.BankHolidayDataClient
+import uk.gov.dluhc.printapi.client.BankHolidayDivision
+import uk.gov.dluhc.printapi.config.DataRetentionConfiguration
+import java.time.DayOfWeek.SATURDAY
+import java.time.DayOfWeek.SUNDAY
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+/**
+ * Responsible for determining the removal date for a Certificate's data. The removal date varies, depending on the
+ * category of data and the type of Certificate concerned.
+ */
+@Component
+class CertificateRemovalDateResolver(
+    private val dataRetentionConfig: DataRetentionConfiguration,
+    private val bankHolidayDataClient: BankHolidayDataClient
+) {
+
+    /**
+     * Calculates the date that the delivery info for a Certificate's print requests should be removed. The retention
+     * period is currently 28 (configurable) working days from the certificate's "issue" date, which means weekends and
+     * bank holidays should be excluded when making this calculation. If the issue date was more than 28 working days
+     * ago, the removal date will be in the past.
+     *
+     * @param issueDate The date the Certificate was issued.
+     * @param gssCode The Certificate's GSS code which is used to determine if there are any upcoming bank holidays for
+     * the UK nation concerned.
+     * @return A [LocalDate] representing when a Certificate's delivery info should be removed
+     */
+    fun getCertificateDeliveryInfoRemovalDate(issueDate: LocalDate, gssCode: String): LocalDate =
```

Comment:
Minor question about the design / use case of this method ....

The kdoc (which is truly excellent BTW 👍 ) refers to a `Certificate` . I wonder whether we mean this in a generic sense, or the entity? - ie the specific type of Certificate (I've read it and assumed the entity, but I could be wrong)

On the assumption that we mean `Certificate` as in the entity, then I think the kdoc should be generalised to describe Certificate as meaning the main certificate for a VAC, but also mention a Temporary Certificate and an AED.

Having said that I wonder if what might be a better approach is if we do actually refer to the entity types, and provide overloaded methods .... 🤔 

We could make this method `getCertificateDeliveryInfoRemovalDate` private, then perhaps have methods for each entity type:
```
fun getPersonalInformationRemovalDate(certificate: Certificate) : Int

fun getPersonalInformationRemovalDate(temporaryCertificate: TemporaryCertificate) : Int

fun getPersonalInformationRemovalDate(aed: AnonymousElectorDocument) : Int // we've not written this entity class yet, but you get the idea)
```

Within those public methods you can get the relevant fields (issueDate and gssCode) from the entity to be able to call the private method

What do you think?

Reply:
It deliberately only refers to Certificate (as in the "business" type, rather than a specific class, though it's also the entity). The kdoc of the class:

```
Responsible for determining the removal date for a Certificate's data. The removal date varies, depending on the category of data and the type of Certificate concerned.
```

Whether "Certificate" is the right term to group these types together is another matter, but I think it works for `Certificate` and `TemporaryCertificate`.

It's difficult to know what lies in store for temp certs & AEDs, so I've not attempted to make this generic, but thought specific methods could be added, or refactoring could be done, as necessary.